### ISSUE TITLE
Add To  Cart Button is not Disbaled  when clicked on this button until product is not add

### ISSUE THREAD

**Issue by Its-Atul** [scandipwa/scandipwa#5583]
Steps to reproduce:

1.Open Any PDP Page
2.Click on Add To Cart Button
3.Wait For Add to Cart To Complete

Actual result

Add To Cart Button is not getting Disabled and User Can Click before previous product add to Cart Completes
Expected result

When Click on Add To Cart Button should be disabled until Operation is completed
#  Testing details

1. isAdding State is not working properly when Aysnc AddtoCart is called
2. For Fallback add to Cart its working as expected

**Comment by Its-Atul** [comment]
Solve My Problem

**Comment by Its-Atul** [comment]
Solution
I have implemented a solution that correctly manages the isAdding state, ensuring that the Add to Cart button disables during the add-to-cart process. If any developer needs guidance on applying this fix, feel free to reach out to me at pandeyatul0310@gmail.com.
https://www.linkedin.com/in/atul-pandey-378a9322a/

Testing Details
After implementing this solution, I tested the Add to Cart button across different product pages and confirmed that it correctly disables until the async operation completes. For fallback add-to-cart operations, the button functions as expected.
